envbuilder.go

```diff
@@ -363,6 +368,17 @@ func Run(ctx context.Context, options Options) error {
 				Password: options.GitPassword,
 			}
 		}
+		if options.GitHttpProxyURL != "" {
+			gitHttpProxyURL, err := url.Parse(options.GitHttpProxyURL)
+			if err != nil {
+				return fmt.Errorf("parse git http proxy url: %w", err)
+			}
+			options.GitHttpProxyURL = gitHttpProxyURL.String()
```

Comment:
I'm confused about why this value is round-tripped through `URL.Parse`. Why parse it into components and then immediately convert back to a string?

Reply:
Just for validation, but will remove it.

Reply:
Thanks. I think it's fine to validate, but you could discard the result of parsing and only validate that it returns no error:

```go
if _, err := url.Parse(options.GitHttpProxyURL); err != nil {
	return fmt.Errorf("invalid git http proxy url: %w", err)
}
```

Doesn't seem necessary, though. What you have now is fine.